source/Missions/Services/Network/Surrogates/Mat3Surrogate.cs

```diff
@@ -0,0 +1,53 @@
+﻿using GameInterface.Serialization.External;
+using GameInterface.Serialization;
+using ProtoBuf;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Library;
+using Common.Serialization;
+using Autofac.Core;
+using Autofac;
+
+namespace Missions.Services.Network.Surrogates
+{
+    [ProtoContract(SkipConstructor = true)]
+    public readonly struct Mat3Surrogate
+    {
+        [ProtoMember(1)]
+        public byte[] data { get; }
```

Comment:
store this as 3 Vec3s instead
```suggestion
        public Vec3 S { get; }
        [ProtoMember(2)]
        public Vec3 F { get; }
        [ProtoMember(3)]
        public Vec3 U { get; }
```